h/views/groups.py

```diff
@@ -112,6 +112,10 @@ def _update_group(self, appstruct):
 def read_unauthenticated(group, request):
     """Group view for logged-out users, allowing them to join the group."""
     check_slug(group, request)
+
+    if group.joinable_by is None:
```

Comment:
I'm not sure I understand the commit message here. Even for logged out users there is a list of effective principals: `['system.Everyone']`. You should be able to use `request.has_permission('join')` to much the same effect?